packages/client/src/workflow-client.ts

```diff
@@ -829,6 +1009,28 @@ export class WorkflowClient extends BaseClient {
     runIdForResult,
     ...resultOptions
   }: WorkflowHandleOptions): WorkflowHandle<T> {
+    // TODO (dan): is there a better location? It must not be visible to users on the WorkflowHandle.
```

Comment:
I think the cleanest solution would be to turn this `WorkflowHandle` into an actual class.

Reply:
OK, yes that's the only thing I could think of also. I vote for doing that in a follow-up PR since the current ad-hoc inline function works, and converting to a class will make non-negligible diff noise.

Reply:
After discussion, this will be addressed by extracting this ad-hoc object to an actual class. This will be done a distinct PR, to avoid conflating this PR.